messaging.md

```diff
@@ -0,0 +1,281 @@
+| Condition Orchestrator | Subscriber |    `com.hollow.sh.serverservice.events.>` |
+
+
+The `Condition Orchestrator`, `Condition Orchestrator API` forwards work for controllers through events containing [`Conditions`](firmware-install-service.md#conditions) for controllers to reconcile (in the above case - collect out of band inventory).
+
+
+| service                    | role       | subject                                   |
+|----------------------------|------------|-------------------------------------------|
+| Condition Orchestrator     | Publisher  | `com.hollow.sh.controllers.commands.>`    |
+| Condition Orchestrator API | Publisher  | `com.hollow.sh.controllers.commands.>`    |
+| Alloy (Controller)         | Subscriber | `com.hollow.sh.controllers.commands.>`    |
+| Flasher (Controller)       | Subscriber | `com.hollow.sh.controllers.commands.>`    |
+| Controllers (all)          | Publisher  | `com.hollow.sh.controllers.replies.>`     |
+
+
+## Event message format
+
+The notes below describe the format of the event messages sent by `subscribers` and `publishers`
+on the metal-toolbox event stream.
+
+In most cases services using the events library [hollow-toolbox events](https://github.com/metal-toolbox/hollow-toolbox/tree/events/events) do not have to deal with the message structure directly. The fields are described here for reference purposes only and users should check the library for more current information.
+
+Events sent using the events library are sent in the [pubsubx.Message](https://github.com/infratographer/x/blob/main/pubsubx/message.go) format, which consists of the below fields.
+
+
+```go
+type Message struct {
+	// SubjectURN is a string representing the identity of the topic of this message
+	SubjectURN string `json:"subject_urn"`
+	// EventType describes the type of event that has triggered this message
+	EventType string `json:"event_type"`
+	// AdditionalSubjectURNs is a group of strings representing additional identities associated with this message
+	AdditionalSubjectURNs []string `json:"additional_subjects"`
+	// ActorURN is a string representing the identity of the actor that created this message
+	ActorURN string `json:"actor_urn"`
+	// Source is a string representing the identity of the source system that created the message
+	Source string `json:"source"`
+	// Timestamp is the time representing when the message was created
+	Timestamp time.Time `json:"timestamp"`
+	// SubjectFields is a map of additional descriptors for this message
+	SubjectFields map[string]string `json:"fields"`
+	// AdditionalData is a field to store any addition information that may be important to include with your message
+	AdditionalData map[string]interface{} `json:"additional_data"`
+}
+```
+
+Here we describe some of the important fields,
+
+1. The `SubjectURN` field is typically in the form of `<namespace>:<resourceType>:<resourceID>`
+
+Where,
+
+ - `<namespace>` is set to `hollow`.
```

Comment:
Like the NATS subjects, defining `namespace` and `resourceType` to constant values seems overly prescriptive. Why is this specified?